Kronos-Shifts-Connector/Microsoft.Teams.Shifts.Integration/Microsoft.Teams.Shifts.Integration.API/Controllers/TeamsController.cs

```diff
@@ -149,11 +153,17 @@ public async Task<ActionResult> UpdateTeam([FromRoute] string aadGroupId)
                 { "IncomingAadGroupId", aadGroupId },
             };
 
+            // get the kronos org job mapping information
+            var teamMappings = await this.teamDepartmentMappingProvider.GetMappedTeamDetailsAsync(aadGroupId).ConfigureAwait(false);
+
+            // get the timezone for the first mapping - logically all mappings for the team must have the same time zone set
+            var timeZone = teamMappings[0].KronosTimeZone;
+
             // Check if payload is for open shift request.
             if (jsonModel.Requests.Any(x => x.Url.Contains("/openshiftrequests/", StringComparison.InvariantCulture)))
             {
                 // Process payload for open shift request.
-                responseModelList = await this.ProcessOpenShiftRequest(jsonModel, updateProps, aadGroupId, isRequestFromCorrectIntegration).ConfigureAwait(false);
+                responseModelList = await this.ProcessOpenShiftRequest(jsonModel, updateProps, aadGroupId, isRequestFromCorrectIntegration, timeZone).ConfigureAwait(false);
```

Comment:
Prefer kronosTimeZone variable name. Please check whole PR for where applicable.

Reply:
Fair enough